test/quantization/test_quant_api.py

```diff
@@ -205,16 +192,21 @@ def test_dynamic_quant_gpu_unified_api_eager_mode_impl(self):
     @unittest.skipIf(not torch.cuda.is_available(), "Need CUDA available")
     def test_int8_wo_quant_save_load(self):
         m = ToyLinearModel().eval().cpu()
-        apply_weight_only_int8_quant(m)
+        m = quantize(m, get_apply_int8wo_quant())
+
+        from torchao.utils import unwrap_tensor_subclass
+        unwrap_tensor_subclass(m)
         example_inputs = m.example_inputs()
         ref = m(*example_inputs)
-        _TMP_FN = "_test.pt"
-        torch.save(m.state_dict(), _TMP_FN)
+        with tempfile.NamedTemporaryFile() as f:
+            torch.save(m.state_dict(), f)
+            f.seek(0)
+            state_dict = torch.load(f)
 
-        state_dict = torch.load(_TMP_FN)
-        os.remove(_TMP_FN)
         m2 = ToyLinearModel().eval()
-        apply_weight_only_int8_quant(m2)
+        m2 = quantize(m2, get_apply_int8wo_quant())
+        unwrap_tensor_subclass(m2)
```

Comment:
why do users need to explicitly do this? This feels clunky

Reply:
yeah this is a temporary workaround for export, also there is some issues with serialization that I'll need to take a look later